ingest/src/main/java/com/microsoft/azure/kusto/ingest/IngestionProperties.java

```diff
@@ -291,22 +291,38 @@ void validate() {
     }
 
     public enum DataFormat {
-        csv,
-        tsv,
-        scsv,
-        sohsv,
-        psv,
-        txt,
-        tsve,
-        json,
-        singlejson,
-        multijson,
-        avro,
-        apacheavro,
-        parquet,
-        orc,
-        raw,
-        w3clogfile
+        csv(IngestionMapping.IngestionMappingKind.Csv, false),
```

Comment:
Isn't this a breaking change?

Reply:
Any existing uses of this enum should work identically. This is only an addition, enabling `getIngestionMappingKind` and `isMappingRequired`.

Reply:
This feels like a breaking change.
Would I be able to switch case on it???
Would I be able to enumerate it?

Why not add a couple of X2Y APIs which take the enum and return the value instead.

Reply:
Yes, all existing uses will work identically.
If by enumerate you're referring to using a hardcoded sequential number representing the order of the item in the enum, in Java that's never assumed unless explicitly instantiated as a property of each enum value. So nobody should do that, like assuming order in a data structure whose order isn't guaranteed.

This approach is OO - the object that supplies the name of the object also supplies the other things it should know about (its `IngestionMappingKind` and `isMappingRequired`).